### PR TITLE
Add support to multiple property pairs in sprint

### PR DESCRIPTION
**Pull request by ronisbr**
Sometimes it is required to pass multiple properties to `IOContext` in
`sprint`. For example, if we want to print with `:compact` and `:limit`
set to true. Currently, the only possible way to do this is creating an
`IOContext` using a dummy `IOBuffer` with those parameters. Hence, this
commit allows to pass a vector of pairs `:key=>value` to `context`
keyword of `sprint` so that we can easily set multiple properties.

This is not a breaking change, and no performance regression was
identified when using the previous function signatures.

### CODE REVIEW

**Comment by ronisbr** [comment]
Bump!

**Comment by simeonschaub** [comment]
I think this should have a compat note in the docstring, but otherwise looks good to me.

**Comment by ronisbr** [comment]
@simeonschaub Done! I think I just need to add the information to NEWS.md, but I will wait for approval to avoid multiple rebases.

**Comment by ronisbr** [comment]
Bump!

File: base/strings/io.jl
Comment (rfourquet) [inline base/strings/io.jl:86]: Not your deed, but isn't it more correct to write
```suggestion
The optional keyword argument `context` can be set to a `:key=>value` pair, a
```
?

File: base/strings/io.jl
Comment (rfourquet) [inline base/strings/io.jl:83]: I think this line should be updated too.

**Review by rfourquet — APPROVED** [review]
LGTM!

**Comment by ronisbr** [comment]
Done @rfourquet ! Thanks for the review.

**Comment by o314** [comment]
Hi there. A pity that namedtuple support has not been implemented in addition.

```julia
# done
sprint(show, v; context=(:compact=>true, :limit=>true))

# still todo
sprint(show, v; context=(; compact=true, limit=true)) # more julian IMHO
```

I'am a bit short on time those days to pr it

**Pull request merged**
